samples/MachineLearning/R/forecast.R

```diff
@@ -344,7 +345,17 @@ smart.forecast <- function(dataset1, dataset2) {
 
   for (i in 1:granularityAttributes_num) {
     # Prepare training and test data for given granularity attribute
-    granularityAttribute_data <- data[which(data$GranularityAttribute == granularityAttributes[i]), ]
+    if (is.na(granularityAttributes[i])) {
+      granularityAttribute_data <- data[which(is.na(data$GranularityAttribute)), ]
```

Comment:
We should also add a test to verify this works and isn't reverted by some other change.

Reply:
Tests are in a separate repository.